Hippo.Web/Controllers/SystemController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Hippo.Core.Data;
+using Hippo.Core.Services;
+using Hippo.Web.Services;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Security.Claims;
+
+namespace Hippo.Web.Controllers
+{
+    public class SystemController : Controller
+    {
+        private readonly AppDbContext _dbContext;
+        private readonly IIdentityService _identityService;
+        private readonly IUserService _userService;
+        public const string IamIdClaimType = "ucdPersonIAMID";
+
+        public SystemController(AppDbContext dbContext, IIdentityService identityService, IUserService userService)
+        {
+            _dbContext = dbContext;
+            _identityService = identityService;
+            _userService = userService;
+        }
+        public async Task<IActionResult> Emulate(string id)
+        {
+            var allowedUsers = new[] {"jsylvest", "postit", "cydoval", "sweber" };
+            var currentUser = await _userService.GetCurrentUser();
+            if(currentUser == null || !allowedUsers.Contains(currentUser.Kerberos))
+            {
+                throw new Exception("Not authorized");

```

Comment:
just an FYI this is fine for emulation but for regular routes in this scenario you'd want to `return NotAuthorized()` instead of throwing an exception

Reply:
@srkirkland I didn't see a NotAuthorized() method, but there is UnAuthorized().